classes/Kohana/Core.php

```diff
@@ -349,7 +349,9 @@ public static function ensureDirectory(string $path, int $permissions): void
         try {
             try {
                 // Create the directory
-                \mkdir($path, $permissions, TRUE);
+                if ( ! mkdir($path, $permissions, TRUE)){
+                    throw new Kohana_Exception('Could not create directory :dir', array(':dir' => Debug::path($path)));
```

Comment:
This all looks good, but while you're here maybe also amend the throw below (in the catch block) to include the original exception message - might as well communicate why it failed if we know? Sorry, I should have done that last time.

Reply:
No worries